invenio_vocabularies/records/models.py

```diff
@@ -53,3 +53,32 @@ class VocabularyMetadata(db.Model, RecordMetadataBase):
     """Model for vocabulary metadata."""
 
     __tablename__ = "vocabularies_metadata"
+
+
+class VocabularySubtype(db.Model):
+    """Vocabulary subtype type model.
+
+    Current use case is for subject types.
+    """
+
+    __tablename__ = "vocabularies_subtypes"
+
+    id = db.Column(db.String, primary_key=True)
```

Comment:
I see the vocabulary_type FK is not part of this table PK. What would happen if there are two subtypes with the same `id` (form different type)?

Reply:
Right now it just means that a different id must be used for a subtype if there is another subtype with desired id but different type :man_shrugging: . In terms of current functionality there is no issue (maybe the current code is not retrieving as fast as it could...), but I am not a DBA. Leaving as is...

Reply:
I was just wondering if we might end up getting into PK collisions, but if not encountered with MeSH/Subjects which is one of the most complex... I guess we will be good :)